webpack/assets/javascripts/react_app/Root/Context/Hooks/useRefreshedContext.js

```diff
@@ -0,0 +1,70 @@
+import axios from 'axios';
+import { useEffect, useState } from 'react';
+import { useForemanSetContext } from '../ForemanContext';
+
+/**
+ * Custom hook that requests the up-to-date application context from the Foreman-backend and updates the ForemanContext accordingly.
+ * Performs an API request to /api/v2/context.
+ * @param {Array<string>} only An array of metadata fields to restrict the context update to.
+ * @returns {{isLoading: boolean, isError: boolean, data: object, error: object, status: number}}
+ */
+const useRefreshedContext = (only = null) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+  const [responseData, setResponseData] = useState(null);
+  const [isError, setIsError] = useState(false);
+  const [status, setStatus] = useState(null);
+
+  const setForemanContext = useForemanSetContext(responseData);
+
+  const getContext = async () => {
+    setIsLoading(true);
+    setIsError(false);
+
+    try {
+      const response = await axios.get('/api/v2/context', {
+        params: { only },
+      });
+      setStatus(response.status);
+      setResponseData(response.data);
+    } catch (err) {
+      setError(err);
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    getContext();
```

Comment:
You can set the getContext inside the useEffect to avoid the error.
But I think instead of getContext and useEffect you can do something like:
``` js
  const { response: responseData, status } = useAPI('get', '/api/v2/context', {
    only,
  });
  const isLoading = status === STATUS.PENDING;
  const isError = status === STATUS.ERROR;
  const error = isError ? responseData : null;

  const setForemanContext = useForemanSetContext(responseData);
  ```
  (also will review the rest as well)